applications/fishing-map/src/features/workspace/activity/ActivitySection.tsx

```diff
@@ -78,22 +88,32 @@ function ActivitySection(): React.ReactElement {
     [dispatchQueryParams, start, end]
   )
 
-  const onAddFishingClick = useCallback(() => {
-    dispatchQueryParams({ bivariateDataviews: undefined })
-    const dataviewInstance = getFishingDataviewInstance()
-    upsertDataviewInstance(dataviewInstance)
-    setAddedDataviewId(dataviewInstance.id)
-  }, [dispatchQueryParams, upsertDataviewInstance])
+  const onAddFishingClick = useCallback(
+    (dataviewId?: number) => {
+      const dataview = fishingDataviews.find((d) => d.id === dataviewId)
+      const dataviewInstance = dataview
+        ? getActivityDataviewInstanceFromDataview(dataview)
+        : getFishingDataviewInstance()
+      if (dataviewInstance) {
+        dispatchQueryParams({ bivariateDataviews: undefined })
+        upsertDataviewInstance(dataviewInstance)
+        setAddedDataviewId(dataviewInstance.id)
+      }
+    },
+    [dispatchQueryParams, fishingDataviews, upsertDataviewInstance]
+  )
 
   const onAddPresenceClick = useCallback(
-    (presenceCategory: WorkspacePresenceCategory) => {
-      dispatchQueryParams({ bivariateDataviews: undefined })
-      const dataviewInstance =
-        presenceCategory === 'presence' ? getPresenceDataviewInstance() : getViirsDataviewInstance()
-      upsertDataviewInstance(dataviewInstance)
-      setAddedDataviewId(dataviewInstance.id)
+    (dataviewId: number) => {
```

Comment:
There's a lot of repetition with `onAddFishingClick`, might be worth factoring the logic?

Reply:
Yes, I created a new function called `addDataviewInstance` to be used in both (Activity and Presence) so we avoid repetition.